webapp/src/components/map/map-slice.tsx

```diff
@@ -0,0 +1,1576 @@
+    let client = Stomp.client(url, protocols)
+    client.debug = (e) => {
+      console.debug('STOMP Debug: ' + e)
+    }
+
+    // Topics are in the format /live/{roadRegulatorID}/{intersectionID}/{spat,map,bsm}
+    let spatTopic = `/live/${roadRegulatorId}/${intersectionId}/spat`
+    let mapTopic = `/live/${roadRegulatorId}/${intersectionId}/map`
+    let bsmTopic = `/live/${roadRegulatorId}/${intersectionId}/bsm` // TODO: Filter by road regulator ID
+    let spatTime = Date.now()
+    let mapTime = Date.now()
+    let bsmTime = Date.now()
+    let connectionStartTime = Date.now()
+    client.connect(
+      {
+        // "username": "test",
+        // "password": "test",
+        // Token: token,
+      },
+      () => {
+        client.subscribe(spatTopic, function (mes: IMessage) {
+          const spatMessage: ProcessedSpat = JSON.parse(mes.body)
+          console.debug('Received SPaT message ' + (Date.now() - spatTime) + ' ms')
+          spatTime = Date.now()
+          dispatch(renderIterative_Spat([spatMessage]))
+          dispatch(maybeUpdateSliderValue())
+        })
+
+        client.subscribe(mapTopic, function (mes: IMessage) {
+          const mapMessage: ProcessedMap = JSON.parse(mes.body)
+          console.debug('Received MAP message ' + (Date.now() - mapTime) + ' ms')
+          mapTime = Date.now()
+          dispatch(renderIterative_Map([mapMessage]))
+          dispatch(maybeUpdateSliderValue())
+        })
+
+        client.subscribe(bsmTopic, function (mes: IMessage) {
+          const bsmData: OdeBsmData = JSON.parse(mes.body)
+          console.debug('Received BSM message ' + (Date.now() - bsmTime) + ' ms')
+          bsmTime = Date.now()
+          dispatch(renderIterative_Bsm([bsmData]))
+          dispatch(maybeUpdateSliderValue())
+        })
+      },
+      (error) => {
+        console.error('Live Streaming ERROR connecting to live data Websocket: ' + error)
+      }
+    )
+
+    client.onDisconnect = (frame) => {
+      console.debug(
+        'Live Streaming Disconnected from STOMP endpoint: ' +
+          frame +
+          ' (numRestarts: ' +
+          numRestarts +
+          ', wsClient: ' +
+          wsClient +
+          ')'
+      )
+      if (numRestarts < 5 && liveDataActive) {
+        let numRestartsLocal = numRestarts
+        if (Date.now() - connectionStartTime > 10000) {
+          numRestartsLocal = 0
+        }
+        console.debug('Attempting to reconnect to STOMP endpoint (numRestarts: ' + numRestartsLocal + ')')
+        dispatch(
+          setLiveDataRestartTimeoutId(
+            setTimeout(() => {
+              dispatch(setLiveDataRestart(numRestartsLocal + 1))
+            }, numRestartsLocal * 2000)
+          )
+        )
+      } else {
+        cleanUpLiveStreaming()
+      }
+    }
+
+    client.onStompError = (frame) => {
+      // TODO: Consider restarting connection on error
```

Comment:
Is this something that should be implemented in this PR?

Reply:
Removed the comment. This callback firing does not signify that the websocket is dead, so should not restart, but can e helpful for debugging other errors